File: R/nutrition/dslc_documentation/functions/preProcessFoodData.R
# Preprocessing function for the food data


# perform pre-processing steps
preProcessFoodData <- function(.food_clean, 
                               .log_transform = TRUE,
                               .center = TRUE,
                               .scale = TRUE,
                               .remove_fat = FALSE) {
  
  food_pre_processed <- .food_clean
  # perform log transformation
  if (.log_transform) {
    food_pre_processed <- food_pre_processed |>
      mutate(across(where(is.numeric), ~log(. + 1)))
  } 
  
  # perform centering
  if (.center) {
    food_pre_processed <- food_pre_processed |>
      mutate(across(where(is.numeric), ~(. - mean(., .na.rm = T))))
  } 
  # perform SD-scaling
  if (.scale) {
    food_pre_processed <- food_pre_processed |>
      mutate(across(where(is.numeric), ~(. / sd(., na.rm = T))))
  }
  
  if (.remove_fat) {
    food_pre_processed <- food_pre_processed |>
      select(-fat)
  }
  
  # do some mean imputation
  food_pre_processed <- food_pre_processed %>%
    mutate(across(where(is.numeric), ~if_else(is.na(.), mean(., na.rm = T), .)))
  
  
  return(food_pre_processed)
}

File: python/nutrition/dslc_documentation/functions/preprocess_food_data.py
# Cleaning function for the food data
import pandas as pd
import numpy as np

# Clean the food data
def preprocess_food_data(food_clean, 
                         log_transform=True,
                         center=True,
                         scale=True,
                         remove_fat=False):
  
  food_pre_processed = food_clean.copy()

  if log_transform:
    food_pre_processed = food_pre_processed.apply(lambda x: np.log10(x + 1) if np.issubdtype(x.dtype, np.number) else x)
  
  
  # perform centering
  if center:
    food_pre_processed = food_pre_processed.apply(lambda x: x - x.mean() if np.issubdtype(x.dtype, np.number) else x)
  
  # perform SD-scaling
  if scale:
    food_pre_processed = food_pre_processed.apply(lambda x: x / x.std() if np.issubdtype(x.dtype, np.number) else x)
  
  if remove_fat:
    food_pre_processed = food_pre_processed.drop(columns="fat")
  
  # do some mean imputation
  food_pre_processed = food_pre_processed.fillna(food_pre_processed.mean())
  
  return food_pre_processed